master/buildbot/changes/bitbucket.py

```diff
@@ -146,7 +146,7 @@ def _processChanges(self, page):
                         comments=u'pull-request #%d: %s\n%s' % (
                             nr, title, prlink),
                         when_timestamp=datetime2epoch(updated),
-                        branch=self.branch,
```

Comment:
You may want to make the destination branch available somewhere so that a ChangeFilter can filter and route changes to the right builder